harness/determined/common/experimental/resource_pool.py

```diff
@@ -0,0 +1,113 @@
+from typing import List, Optional
+
+from determined.common import api
+from determined.common.api import bindings
+from determined.common.experimental import workspace
+
+
+class ResourcePool:
+    """A class representing a resource pool object.
+
+    Attributes:
+        name: (str) The name of the resource pool.
+    """
+
+    def __init__(
+        self,
+        session: api.Session,
+        name: Optional[str] = None,
+    ):
+        """Create a resource pool object.
+
+        Arguments:
+            session: The session to use for API calls.
+            workspace_name: (Optional) Name of the resource pool.
+        """
+        self.name = name
+        self._session = session
+
+    @classmethod
+    def _from_bindings(
+        cls, resource_pool_bindings: bindings.v1ResourcePool, session: api.Session
+    ) -> "ResourcePool":
+        resource_pool = cls(session, name=resource_pool_bindings.name)
+        return resource_pool
+
+    def bind(
+        self,
+        workspace_names: List[str],
+    ) -> None:
```

Comment:
We've been inconsistent within the codebase and I don't know how you could have known otherwise, but please follow the Google style guide (itself a more detailed specification than PEP-8) [for docstrings](https://google.github.io/styleguide/pyguide.html#383-functions-and-methods).

It's a lot of text to read, but it works pretty well to just crib off the examples.